toolchain/check/name_component.cpp

```diff
@@ -11,76 +11,63 @@ namespace Carbon::Check {
 
 auto PopNameComponent(Context& context, SemIR::InstId return_slot_pattern_id)
     -> NameComponent {
-  Parse::NodeId first_param_node_id = Parse::InvalidNodeId();
-  Parse::NodeId last_param_node_id = Parse::InvalidNodeId();
+  NameComponent name_component = {
+      .name_loc_id = Parse::NodeId::Invalid,
+      .name_id = SemIR::NameId::Invalid,
+      .first_param_node_id = Parse::NodeId::Invalid,
+      .last_param_node_id = Parse::NodeId::Invalid,
+      .implicit_params_loc_id = Parse::NodeId::Invalid,
+      .implicit_param_patterns_id = SemIR::InstBlockId::Invalid,
+      .params_loc_id = Parse::NodeId::Invalid,
+      .param_patterns_id = SemIR::InstBlockId::Invalid,
+      .call_params_id = SemIR::InstBlockId::Invalid,
+      .return_slot_pattern_id = return_slot_pattern_id,
+      .pattern_block_id = SemIR::InstBlockId::Invalid,
+  };
 
   // Explicit params.
-  auto [params_loc_id, param_patterns_id] =
-      context.node_stack().PopWithNodeIdIf<Parse::NodeKind::TuplePattern>();
-  if (param_patterns_id) {
-    first_param_node_id =
+  if (auto [params_loc_id, param_patterns_id] =
+          context.node_stack().PopWithNodeIdIf<Parse::NodeKind::TuplePattern>();
+      param_patterns_id) {
+    name_component.first_param_node_id =
         context.node_stack()
             .PopForSoloNodeId<Parse::NodeKind::TuplePatternStart>();
-    last_param_node_id = params_loc_id;
-  } else {
-    param_patterns_id = SemIR::InstBlockId::Invalid;
+    name_component.last_param_node_id = params_loc_id;
```

Comment:
We used to set `last_param_node_id` and `param_patterns_id` if there _was_ a TuplePattern, which made sense to me, the TuplePattern gives us a pattern. But now we're setting these if there's _not_ a TuplePattern, how come?

Reply:
This was me misreading the unified diff, where the else was above here. It's still in the same branch, actually.